File: app/src/main/java/de/euhm/jlt/CustomApplication.java
/**
 * @file CustomApplication.java
 * 
 * Licensed under the Apache License, Version 2.0 (the "License")
 * http://www.apache.org/licenses/LICENSE-2.0
 */
package de.euhm.jlt;

import android.app.Application;

/**
 * Custom application to keep track of the activity visibility
 * @author hmueller
 */
public class CustomApplication extends Application {

	  private static boolean mActivityVisible;
	  
	  public static boolean isActivityVisible() {
	    return mActivityVisible;
	  }  

	  public static void activityResumed() {
	    mActivityVisible = true;
	  }

	  public static void activityPaused() {
	    mActivityVisible = false;
	  }
}

File: app/src/main/java/de/euhm/jlt/CustomApplication.kt
/**
 * @file CustomApplication.kt
 *
 * Converted from Java to Kotlin by Android Studio Meerkat | 2024.3.1 in March 2025
 *
 * MIT License
 * Copyright (c) 2014-2025 Holger Mueller
 */
package de.euhm.jlt

import android.app.Application

/**
 * Custom application to keep track of the activity visibility
 * @author hmueller
 */
object CustomApplication : Application() {
    var isActivityVisible: Boolean = false
        private set

    fun activityResumed() {
        isActivityVisible = true
    }

    fun activityPaused() {
        isActivityVisible = false
    }
}
